frontend/src/components/ItemCardList.tsx

```diff
@@ -39,7 +40,7 @@ const ItemCardList = ({
                     href={item?.url}
                     target="_blank"
                   >
-                    {item.title || item.name}
+                    <TruncatedText text={item.title || item.name} maxLength={40} />
```

Comment:
This approach is not ideal as the solution has to be universal for desktop/mobile screens (please read the issue description). 

See examples below:
<img width="577" alt="Screenshot 2025-03-15 at 3 33 02 PM" src="https://github.com/user-attachments/assets/7e0c42f6-c822-4041-8a5a-74c88ebdaaa8" />
<img width="567" alt="Screenshot 2025-03-15 at 3 33 12 PM" src="https://github.com/user-attachments/assets/bb3690d9-b827-4cbf-b86a-b65a2fadd5f8" />
<img width="1145" alt="Screenshot 2025-03-15 at 3 33 18 PM" src="https://github.com/user-attachments/assets/71f94824-d248-495c-ba2a-6ab38340c761" />